### PR TITLE
Refactor serializer task to break into individual reusable serializers

### PR DESCRIPTION
**Pull request by neakor**
This allows the pluginized serialization task to reuse individual serializers.

### CODE REVIEW

File: Generator/Sources/NeedleFramework/Generating/Serializers/DependencyProviderInitBodySerializer.swift [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import Foundation
+
+/// A serializer that produces the initializer body code for the dependency
+/// provider.
+class DependencyProviderInitBodySerializer: Serializer {
+
+    /// Initializer.
+    ///
+    /// - parameter provider: The provider to generate initializer body
+    /// code for.
```
Comment (rudro) [inline Generator/Sources/NeedleFramework/Generating/Serializers/DependencyProviderInitBodySerializer.swift]: nit: I'd say `source code` as you do below.

File: Generator/Sources/NeedleFramework/Generating/Serializers/DependencyProviderRegistrationSerializer.swift [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+    /// Initializer.
+    ///
+    /// - parameter provider: The provider to generate registration code
+    /// for.
+    init(provider: ProcessedDependencyProvider) {
+        self.provider = provider
+    }
+
+    /// Serialize the data model and produce the registration code.
```
Comment (rudro) [inline Generator/Sources/NeedleFramework/Generating/Serializers/DependencyProviderRegistrationSerializer.swift]: nit: `source code`

File: Generator/Sources/NeedleFramework/Generating/Serializers/DependencyProviderSerializer.swift [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    init(provider: ProcessedDependencyProvider, classNameSerializer: Serializer, propertiesSerializer: Serializer, sourceComponentsSerializer: Serializer, initBodySerializer: Serializer) {
+        self.provider = provider
+        self.classNameSerializer = classNameSerializer
+        self.propertiesSerializer = propertiesSerializer
+        self.sourceComponentsSerializer = sourceComponentsSerializer
+        self.initBodySerializer = initBodySerializer
+    }
+
+    /// Serialize the data model and produce the entire dependency provider
+    /// code.
```
Comment (rudro) [inline Generator/Sources/NeedleFramework/Generating/Serializers/DependencyProviderSerializer.swift]: nit: same here and 2 lines below.

File: Generator/Sources/NeedleFramework/Generating/Serializers/DependencyProviderInitBodySerializer.swift (hunk 1 above)
Comment (neakor) [reply Generator/Sources/NeedleFramework/Generating/Serializers/DependencyProviderInitBodySerializer.swift]: Done.

File: Generator/Sources/NeedleFramework/Generating/Serializers/DependencyProviderRegistrationSerializer.swift (hunk 2 above)
Comment (neakor) [reply Generator/Sources/NeedleFramework/Generating/Serializers/DependencyProviderRegistrationSerializer.swift]: Done.

File: Generator/Sources/NeedleFramework/Generating/Serializers/DependencyProviderSerializer.swift (hunk 3 above)
Comment (neakor) [reply Generator/Sources/NeedleFramework/Generating/Serializers/DependencyProviderSerializer.swift]: Done.

**Pull request merged**